### ISSUE TITLE
How to change the edge_color after the object is created

### ISSUE THREAD

**Issue by zhu2900000** [vispy/vispy#2381]
hi~

```
import sys

from vispy import scene
from vispy.color import Color
canvas = scene.SceneCanvas(keys='interactive', size=(800, 600), show=True)

# Set up a viewbox to display the cube with interactive arcball
view = canvas.central_widget.add_view()
view.bgcolor = '#efefef'
view.camera = 'turntable'
view.padding = 1

color = Color(color='#3f51b5', alpha=0.5, clip=True)

cube1 = scene.visuals.Box(1, 1, 1, color=color, edge_color="black",
                         parent=view.scene)
cube1.edge_color = "green"
# cube.size[1]=100
if __name__ == '__main__' and sys.flags.interactive == 0:
    canvas.app.run()
```

```
raise AttributeError('%r is not an attribute of class %s. Call '
AttributeError: 'edge_color' is not an attribute of class <Box at 0x7f8188c2da90>. Call "unfreeze()" to allow addition of new attributes
```

**Comment by djhoese** [comment]
I think `cube1.border.color` might work. Yeah that is not intuitive, sorry.

**Comment by zhu2900000** [comment]
> I think `cube1.border.color` might work. Yeah that is not intuitive, sorry.
It real works thanks.

**Comment by djhoese** [comment]
@brisvag I kind of feel like we should change/add `border_color` or add a new `border_kwargs` or something option to make this a little more clear. Thoughts?

**Comment by brisvag** [comment]
Yeah, Most straightforward I think is to just have a property called edge_color.

**Comment by djhoese** [comment]
You mean `border_color` and then it is accessed via `.border.color`. It is currently called `edge_color`. We have a lot of Visuals like this that are CompoundVisuals with sub-visuals in them. Some have top level properties that access the lower-level object for the user `my_compound_vis.border_color` will do `self._border.color = color` but others expect `self.border.color = color`. How we pass arguments to `__init__` and modify those properties later is something we should really figure out. Things get more confusing in cases where a Visual could probably optimize something like a MeshVisual to serve two purposes (by combining multiple configurations in a single set of arrays).

**Comment by brisvag** [comment]
I think I'm following some of what you say :P But yeah, I agree we should be more consistent between accessors; my feeling is that in general we should strive to have the same names for attributes/setters and arguments for init/set_data.
